crates/project_panel/src/project_panel.rs

```diff
@@ -694,10 +694,12 @@ impl ProjectPanel {
 
         let worktree_id = edit_state.worktree_id;
         let is_new_entry = edit_state.is_new_entry;
+        let filename = self.filename_editor.read(cx).text(cx);
+        edit_state.is_dir =
+            edit_state.is_dir || (edit_state.is_new_entry && filename.ends_with('/'));
```

Comment:
```suggestion
            edit_state.is_dir || (edit_state.is_new_entry && filename.ends_with(std::path::MAIN_SEPARATOR));
```

Let's not forget about Windows.

Reply:
I cannot test it on windows but shouldn't it be `filename.ends_with(std::path::MAIN_SEPARATOR) || filename.ends_with('/')` then ?

The question is that if we allow both `/` and `\` as path separators on windows or only `\` ?

Reply:
It's just `std::path::MAIN_SEPARATOR` which will be different depending on the platform it's compiled to.

Reply:
No tests are run on Windows yet, and I expect there are bugs.

> `filename.ends_with(std::path::MAIN_SEPARATOR) || filename.ends_with('/')` 

is never right, as `std::path::MAIN_SEPARATOR` is already `\` when compiled to Unix, and `/` to Windows targets.
Eventually, we'll have to fix all path-related issues and support both, but without direct manipulation of `/` or `\` symbols.